File: solutions/0442-find-all-duplicates-in-an-array/solution.py
class Solution(object):
    def findDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ls={}
        ls1=[]
        for i in nums:
            if i in ls:
                ls[i]+=1
            else:
                ls[i]=1
            if ls[i]==2:
                ls1.append(i)
        return ls1

File: solutions/0442-find-all-duplicates-in-an-array/solution.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> ls=new ArrayList<>();
        int n=nums.length;
        for(int i=0;i<n;i++)
        {
            int id = Math.abs(nums[i]) - 1;
            if(nums[id]<0) 
                ls.add(Math.abs(nums[i]));
            else
                nums[id]=-nums[id];
        }
        return ls;
    }
}
